src/cameraGeom/TransformMap.cc

```diff
@@ -84,12 +88,160 @@ std::shared_ptr<ast::Mapping const> TransformMap::_getMapping(CameraSys const &f
+    }
+
+    table::Schema sysSchema;
+    table::Key<std::string> sysName;
+    table::Key<std::string> detectorName;
+    table::Key<int> id;
+
+    table::Schema connectionSchema;
+    table::Key<int> from;
+    table::Key<int> to;
+    table::Key<int> transform;
+
+    CameraSys makeCameraSys(table::BaseRecord const & record) const {
+        return CameraSys(record.get(sysName), record.get(detectorName));
+    }
+
+private:
+
+    PersistenceHelper() :
+        sysSchema(),
+        sysName(sysSchema.addField<std::string>("sysName", "Camera coordinate system name", "", 0)),
+        detectorName(sysSchema.addField<std::string>("detectorName",
+                                                     "Camera coordinate system detector name", "", 0)),
+        id(sysSchema.addField<int>("id", "AST ID of the Frame for the CameraSys", "")),
+        connectionSchema(),
+        from(connectionSchema.addField<int>("from", "AST ID of the Frame this transform maps from.", "")),
+        to(connectionSchema.addField<int>("to", "AST ID of the Frame this transform maps to.", "")),
+        transform(connectionSchema.addField<int>("transform", "Archive ID of the transform.", ""))
+    {
+        sysSchema.getCitizen().markPersistent();
+        connectionSchema.getCitizen().markPersistent();
+    }
+
+    PersistenceHelper(PersistenceHelper const &) = delete;
+    PersistenceHelper(PersistenceHelper &&) = delete;
+
+    PersistenceHelper & operator=(PersistenceHelper const &) = delete;
+    PersistenceHelper & operator=(PersistenceHelper &&) = delete;
+
+};
+
+std::string makeFrameName(CameraSys const & sys) {
+    std::string r = "Ident=" + sys.getSysName();
+    if (sys.hasDetectorName()) {
+        r += "_";
+        r += sys.getDetectorName();
+    }
+    return r;
+}
+
+} // anonymous
+
+
+std::string TransformMap::getPersistenceName() const {
+    return "TransformMap";
+}
+
+std::string TransformMap::getPythonModule() const {
+    return "lsst.afw.cameraGeom";
+}
+
+void TransformMap::write(OutputArchiveHandle& handle) const {
+    auto const & keys = PersistenceHelper::get();
+
+    auto sysCat = handle.makeCatalog(keys.sysSchema);
+    for (auto const & sysPair : _frameIds) {
+        auto sysRecord = sysCat.addNew();
+        sysRecord->set(keys.sysName, sysPair.first.getSysName());
+        sysRecord->set(keys.detectorName, sysPair.first.getDetectorName());
+        sysRecord->set(keys.id, sysPair.second);
+    }
+    sysCat.sort(keys.id);
+    handle.saveCatalog(sysCat);
+
+    auto connectionCat = handle.makeCatalog(keys.connectionSchema);
+    for (auto const & connectionPair : _canonicalConnections) {
+        auto connectionRecord = connectionCat.addNew();
+        connectionRecord->set(keys.from, connectionPair.first);
+        connectionRecord->set(keys.to, connectionPair.second);
+        auto transform = geom::TransformPoint2ToPoint2(
+            *_transforms->getMapping(connectionPair.first, connectionPair.second)
+        );
```

Comment:
weird formatting imo

Reply:
Huh.  I actually really like this form; it's the only way to get a really long single-argument in that I find readable (and it seems the style guide is silent on layout for *calling* functions).

I'm guessing you'd prefer just putting the closing `);` on the previous line, or were you expecting something else entirely?  (I think I must just have a strong aversion to not being able to find closing braces/brackets/parens).